openmc/mgxs_library.py

```diff
@@ -457,61 +366,173 @@ def use_chi(self, use_chi):
         check_type('use_chi', use_chi, bool)
         self._use_chi = use_chi
 
-    @total.setter
-    def total(self, total):
+    def add_temperature(self, temperature):
+        """This method re-sizes the attributes of this XSdata object so that it
+        can accomodate an additional temperature.  Note that the set_* methods
+        will still need to be executed.
+
+        Parameters
+        ----------
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset.
+
+        """
+        check_type('temperature', temperature, Real)
+
+        temp_store = self.temperatures.tolist().append(temperature)
+        self.temperatures = temp_store
+
+        self._total.append(None)
+        self._absorption.append(None)
+        self._scatter_matrix.append(None)
+        self._multiplicity_matrix.append(None)
+        self._fission.append(None)
+        self._nu_fission.append(None)
+        self._kappa_fission.append(None)
+        self._chi.append(None)
+
+    def set_total(self, total, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        total: nd.nparray
+            Total Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_total_mgxs()
+
+        """
         check_type('total', total, Iterable, expected_iter_type=Real)
         # Convert to a numpy array so we can easily get the shape for
         # checking
```

Comment:
Can you move this (and other instances of this comment) to the previous line? I think it will still be under the 80 character limit.